### DISCUSSION TITLE
Add bridge to yandex smart home

### DISCUSSION THREAD

**Discussion by Nik768** [Luligu/matterbridge#48] [Q&A]
Hello, I need a consultation. I use yandex smart home as a voice control smart home system, because it is best adapted for my language (russian). However, no matter how I tried to add a bridge there, I can't.
In HA, this bridge is thrown through without any problems. My question is not about how to fix it. Since these are devices common only in the CIS, it is unlikely that something will be fixed if it is connected with the bridge. I have a log when I try to attach a bridge, I'm just wondering if yandex is blocking the connection or is it somehow related to the bridge itself?
I also tried adding a bridge to the Aqara Home, just to test. He writes that an exception occurs when checking the certificate, but log in metterbridge looks different.  And i tried node red matter, the situation was similar.
Thanks!

`2024-06-12 12:04:35.889 INFO   PaseServer           Received pairing request from udp://fe80::3e0b:4fff:fec7:4e04%enp0s3:36781.
2024-06-12 12:04:35.954 INFO   PaseServer           Session 44973 created with udp://fe80::3e0b:4fff:fec7:4e04%enp0s3:36781.
2024-06-12 12:04:35.964 INFO   SessionManager       End insecure session insecure/10403888681497528080
2024-06-12 12:04:36.043 INFO   MatterDevice         Announce done fabric: 1 fabricIndex: 1
2024-06-12 12:04:36.044 INFO   AttributeServer      Set remote value for fabric scoped attribute "acl" to [{"fabricIndex":1,"privilege":5,"authMode":2,"subjects":["1"],"targets":null}]
2024-06-12 12:04:36.045 INFO   LegacyInte~ionServer ACL updated { fabricIndex: 1, privilege: 5, authMode: 2, subjects: [ 1 ], targets: null }
2024-06-12 12:04:36.045 INFO   Operationa~alsServer addNoc success, adminVendorId 5130, caseAdminSubject 1
2024-06-12 12:04:36.253 INFO   CaseServer           Received pairing request from udp://fe80::3e0b:4fff:fec7:4e04%enp0s3:36781
[12:04:36.268] [Matterbridge] Active session changed on fabric 1 id 1 vendor 5130 (unknown)  for Matterbridge { name: 'secure/44974', nodeId: 294, peerNodeId: 1, fabric: { fabricIndex: 1, fabricId: 1, nodeId: 294, rootNodeId: 1, rootVendorId: 5130, label: '' }, isPeerActive: true, secure: true, lastInteractionTimestamp: 1718193876268, lastActiveTimestamp: 1718193876268, numberOfActiveSubscriptions: 0 }
2024-06-12 12:04:36.269 INFO   CaseServer           session 44974 created with udp://fe80::3e0b:4fff:fec7:4e04%enp0s3:36781 for Fabric 0000000000000126(index 1) and PeerNode 0000000000000001
2024-06-12 12:04:36.291 INFO   MatterDevice         All announcements expired
2024-06-12 12:04:36.291 INFO   SecureSession        Register Session secure/44973 to send a close when exchange is ended.
2024-06-12 12:04:36.292 INFO   GeneralCom~erHandler Commissioning completed on fabric #1 as node #294.
2024-06-12 12:04:36.293 INFO   SessionManager       End insecure session insecure/17403569245958842507
2024-06-12 12:04:36.315 INFO   SecureSession        Register Session secure/44974 to send a close when exchange is ended.
2024-06-12 12:04:36.315 INFO   MatterDevice         All announcements expired
2024-06-12 12:04:36.323 ERROR  InteractionServer    Subscription 1334956644 for Session 44974: Error while sending initial data reports EndOfStreamError
    at Queue.close (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/util/Queue.js:51:31)
    at MessageExchange.closeInternal (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/MessageExchange.js:377:24)
    at MessageExchange.destroy (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/MessageExchange.js:318:16)
    at ExchangeManager.closeSession (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/ExchangeManager.js:236:24)
    at async ExchangeManager.deleteExchange (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/ExchangeManager.js:217:9)
    at async MessageExchange.closeCallback (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/ExchangeManager.js:169:23)
    at async MessageExchange.closeInternal (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/MessageExchange.js:378:5)
    at async InteractionServerMessenger.handleRequest (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/interaction/InteractionMessenger.js:165:7)
    at async LegacyInteractionServer.onNewExchange (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/interaction/InteractionServer.js:92:5)
    at async ExchangeManager.onMessage (file:///home/sochan/.local/share/fnm/node-versions/v20.14.0/installation/lib/node_modules/matterbridge/node_modules/@project-chip/matter.js/dist/esm/protocol/ExchangeManager.js:173:9)
2024-06-12 12:04:36.325 INFO   ExchangeManager      Exchange with index 36451 to delete not found or already deleted.
2024-06-12 12:04:36.325 INFO   ExchangeManager      Exchange with index 36451 to delete not found or already deleted.
2024-06-12 12:04:36.325 INFO   SessionManager       End CASE session secure/44974
2024-06-12 12:04:36.441 INFO   ExchangeManager      Exchange with index 89603 to delete not found or already deleted.
2024-06-12 12:04:36.441 INFO   SessionManager       End PASE session secure/44973
2024-06-12 12:04:36.445 INFO   CommissioningServer  The device was factory reset and restarted.
[12:04:36.446] [Matterbridge] Commissioning removed from fabric 1 for Matterbridge. Resetting the commissioning server ...
2024-06-12 12:04:36.531 INFO   MatterDevice         All announcements expired
2024-06-12 12:04:36.536 INFO   CommissioningServer  The device was factory reset and restarted.`

**Comment by Luligu** [comment]
Hi,
is difficult to say but it seems more like network errors... 
Maybe try to connect with network cable both the controller and the machine where you run Matterbridge.
As of my knowledge all controllers pairs with Matterbridge/matter.js
Some of them are more difficult... try again after rebooting routers access points...

**Reply by Apollon77** [reply]
The error seems to happen during subscription request. Would be cool to see a debug log. But the pairing itself seems to was suscccessful

**Comment by alexxx113** [comment]
the same problem.
log by link
[Log](https://github.com/alexxx113/musor/blob/main/log_matter.txt)

**Reply by Apollon77** [reply]
Please enable debug logging and send debug log.
In general it looks like the device did not correctly respond to the device sending the initial data for the subscription.
So debug log should shed some more light

Is it possible to try it too? Is there any hub needed from yandex or just an App? Which?

**Reply by alexxx113** [reply]
thanks for the answer!
We need a yandex speaker. (
how do I launch a debug with addon home assistant ?

**Reply by Nik768** [reply]
Hi. I'll probably go back to the discussion and send a log from my side. I have a feeling that this is an artificial restriction on the connection of external devices on the part of Yandex, but unfortunately the Yandex Smart Home application does not provide any information about this. It is also important to note that the list of supported matter devices is limited to sockets and LED strips https://alice.yandex.ru/support/ru/smart-home/supported-matter-devices (page only in russian)
Therefore, I do not add anything to the matter bridge whitelist except sockets for checking.
However, in my experience, I can say that he calmly found devices with the AqaraHub M3, so most likely he also supports bridges in some form.
Yandex station Midi (last model) support local voice control of matter and zigbee devices and if it will work we can connect homeassistant and yandex station in local network with local voice control.
Now I have to use AqaraHub M3 to use zigbee Aqara devices simultaneously in Homeassistant and Yandex Smart Home at the same time via matter, not the best way, but it works.

> Is it possible to try it too? Is there any hub needed from yandex or just an App? Which?

Unfortunately, it needs a hub in the form of a Yandex station (smart speaker)

> Please enable debug logging and send debug log.

[matterbridge_to_yandex_sm.log](https://github.com/user-attachments/files/16323925/matterbridge_to_yandex_sm.log)

**Reply by Apollon77** [reply]
Ok ... so this log is interesting ..... 
* it does a commissioning ... all successful
*  it reads network interfaces
* it subscribes for changes to the product id ... this subscription in fact never completes because it do not get any response from yandex part! (there somes an stand alone ack message but thats not the expected response in this case
* in parallel to last it reads his fabric index and then triggers a removeFabric to remove itself
* this removal is done and thats the reason also why the subscription which was "In flight" is cancelled and gives this error

So to summarize: It is a yandex issue ... please report it to them, maybe lead them to me, if anyone has a contact (linked in, email or such) then they can contact me or tell and I will try to reach ... I will see if I find yandex responsible persons in my channels

**Reply by Nik768** [reply]
@Apollon77 It's sad, but thank you!
